### PR TITLE
feat(action): register GCP service account

### PR DESCRIPTION
**Pull request by uwehdaub**
To use the kustomize plugin to read secrets from
Google Secret Manager and Service Account key
needs to be registered.

### CODE REVIEW

**Review by PatrykBruder — APPROVED** [review]
LGTM 👍

File: src/gcp-sa.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as core from '@actions/core'
+import { promises as fs } from 'fs'
+import path from 'path'
+import { v4 as uuidv4 } from 'uuid'
+
+export async function setupGcpSa (serviceAccountKey: string): Promise<void> {
```
Comment (navyatejaotturu-metronom) [inline src/gcp-sa.ts]:
```suggestion
export async function setupServiceAccount (serviceAccountKey: string): Promise<void> {
```

File: src/gcp-sa.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  try {
+    const credsDir = String(process.env.GITHUB_WORKSPACE)
+    const credsPath = path.join(credsDir, uuidv4())
+
+    const serviceAccountKeyObj = parseServiceAccountKey(
```
Comment (navyatejaotturu-metronom) [inline src/gcp-sa.ts]:
```suggestion
    const serviceAccount = parseServiceAccountKey(
```

File: src/gcp-sa.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      serviceAccountKey
+    )
+
+    await fs.writeFile(credsPath, JSON.stringify(serviceAccountKeyObj, null, 2))
```
Comment (navyatejaotturu-metronom) [inline src/gcp-sa.ts]:
```suggestion
    await fs.writeFile(credsPath, JSON.stringify(serviceAccount, null, 2))
```

File: src/gcp-sa.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    core.exportVariable('GCLOUD_PROJECT', serviceAccountKeyObj.project_id)
```
Comment (navyatejaotturu-metronom) [inline src/gcp-sa.ts]:
```suggestion
    core.exportVariable('GCLOUD_PROJECT', serviceAccount.project_id)
```

File: src/main.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import * as core from '@actions/core'
 
 import * as config from './config'
 import * as install from './install'
+import * as gcp from './gcp-sa'
```
Comment (navyatejaotturu-metronom) [inline src/main.ts]: Why not to name the file also as `gcp` instead of `gcp-sa`?

File: src/main.ts [hunk 6]
```diff
@@ -11,6 +12,11 @@ export async function run (): Promise<void> {
     return
   }
 
+  const serviceAccountKey = core.getInput('gcp_sa_key')
+  if (serviceAccountKey) {
+    await gcp.setupGcpSa(serviceAccountKey)
```
Comment (navyatejaotturu-metronom) [inline src/main.ts]:
```suggestion
    await gcp.setupServiceAccount(serviceAccountKey)
```
It stutter if we use `gcp` twice. Hence `gcp.setupServiceAccount`.

**Review by navyatejaotturu-metronom — CHANGES_REQUESTED** [review]
It looks good. I feel like we need these simple name changes so that the code is a bit more readable and understandable.

**Comment by navyatejaotturu-metronom** [comment]
I have used `npm outdated` command and see quite a few outdated dependencies. I think we can take this chance and update them also.
```
Package                           Current   Wanted   Latest  Location
@types/jest                       26.0.23  26.0.24  26.0.24  setup-tools-for-waas
@types/js-yaml                      4.0.1    4.0.2    4.0.2  setup-tools-for-waas
@types/node                       15.12.2  15.14.7  16.4.13  setup-tools-for-waas
@types/semver                       7.3.6    7.3.8    7.3.8  setup-tools-for-waas
@typescript-eslint/eslint-plugin   4.28.5   4.29.0   4.29.0  setup-tools-for-waas
@typescript-eslint/parser          4.28.5   4.29.0   4.29.0  setup-tools-for-waas
eslint-plugin-github                4.1.3    4.1.5    4.1.5  setup-tools-for-waas
eslint-plugin-import               2.23.4   2.24.0   2.24.0  setup-tools-for-waas
eslint-plugin-jest                 24.3.6   24.4.0   24.4.0  setup-tools-for-waas
jest                               27.0.4   27.0.6   27.0.6  setup-tools-for-waas
jest-circus                        27.0.4   27.0.6   27.0.6  setup-tools-for-waas
ts-jest                            27.0.3   27.0.4   27.0.4  setup-tools-for-waas
```

File: src/main.ts (hunk 5 above)
Comment (uwehdaub) [reply src/main.ts]: changed

**Pull request merged**